server/broadcast.go

```diff
@@ -233,18 +232,33 @@ func processSegment(cxn *rtmpConnection, seg *stream.HLSSegment) {
 		}
 	}
 
+	// Process the rest of the segment asynchronously - transcode
+	go func() {
+		for true {
+			// if fails, retry; rudimentary
+			if err := transcodeSegment(cxn, seg, name); err == nil {
+				return
+			}
+		}
+	}()
+}
+
+func transcodeSegment(cxn *rtmpConnection, seg *stream.HLSSegment, name string) error {
+
+	nonce := cxn.nonce
+	rtmpStrm := cxn.stream
+	cpl := cxn.pl
 	sess := cxn.sessManager.selectSession()
 	// Return early under a few circumstances:
-	// View-only (non-transcoded) streams or mid-failover
+	// View-only (non-transcoded) streams or no sessions available
 	if sess == nil {
 		if monitor.Enabled {
 			monitor.LogSegmentTranscodeFailed(monitor.SegmentTranscodeErrorNoOrchestrators, nonce, seg.SeqNo, errors.New("No Orchestrators Error"))
 		}
-		return
+		return nil
 	}
+	{
```

Comment:
This is to save the indentation from gofmt in order to make the actual differences in this commit easier to eyeball. We can do a separate commit for cosmetics later.